code/weapon/weapons.cpp

```diff
@@ -785,48 +790,83 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		create_if_not_found = false;
 	}
 
+	// check first if this is on the remove blacklist
+	auto it = std::find(Removed_weapons.begin(), Removed_weapons.end(), fname);
+	if (it != Removed_weapons.end()) {
+		remove_weapon = true;
+	}
+	
+	// we might have a remove tag
+	if (optional_string("+remove")) {
+		if (!replace) {
+			Warning(LOCATION, "+remove flag used for weapon in non-modular table");
+		}
+		if (!remove_weapon) {
+			Removed_weapons.push_back(fname);
+			remove_weapon = true;
+		}
+	}
+
 	//Remove @ symbol
 	//these used to be used to denote weapons that would
 	//only be parsed in demo builds
 	if ( fname[0] == '@' ) {
 		backspace(fname);
 	}
 
+	//Check if weapon exists already
 	int w_id = weapon_info_lookup(fname);
 
-	if(w_id != -1)
+	// maybe remove it
+	if (remove_weapon)
+	{
+		if (w_id >= 0) {
+			mprintf(("Removing previously parsed weapon '%s'\n", fname));
+			Weapon_info.erase(Weapon_info.begin() + w_id);
```

Comment:
That line scares me :)
maybe add an Assert, that `w_id < weapon_info.size()`

Reply:
That isn't needed because a) `weapon_info_lookup()` will never return a number that high, b) that is already enforced by range checks inside `std::vector.erase()`.

Reply:
uh, i didn't know about the check in erase.
thats nice